src/utils.js

```diff
@@ -31,31 +31,37 @@ var csrfSettings = {
 };
 
 function fetchHeaders( opts ) {
-  const {method = 'get', dataType, contentType='application/vnd.api+json'} = opts || {};
+  const {method = 'get', dataType, contentType='application/vnd.api+json', additionalHeaders} = opts || {};
   let headers = new Headers({
     'X-Requested-With': 'XMLHttpRequest'
   });
   if( dataType == 'json' )
     headers.set( 'Content-Type', contentType );
   if( !(/^(GET|HEAD|OPTIONS\TRACE)$/i.test( method )) )
     headers.set( 'X-CSRFToken', csrfSettings.token );
+  for ( const k in additionalHeaders )
+    headers.set(k, additionalHeaders[k])
   return headers;
 }
 
-export function ajax( url, body, method, dataType, contentType ) {
+export function ajax( url, body, method, dataType, contentType, additionalHeaders ) {
   let requestInit = {
     method,
-    headers: fetchHeaders( {method, dataType, contentType} ),
+    headers: fetchHeaders( {method, dataType, contentType, additionalHeaders} ),
     credentials: 'same-origin'
   };
-  if( method.toLowerCase() != 'get' &&  method.toLowerCase() != 'head' )
+  if( method.toLowerCase() != 'get' &&  method.toLowerCase() != 'head' && method.toLowerCase() != 'options')
     requestInit.body = body
   let request = new Request( url, requestInit );
   return fetch( request )
     .then( response => {
       if( response.ok ) {
-        if( response.status != 204 )
-          return response.json();
+        if( response.status != 204 ) {
+          if (TINYAPI_NODE) {
```

Comment:
What's the need for the `TINYAPI_NODE` flag?

Reply:
If it's running in Node then we may need to deal with the cookies etc. manually, so this returns the full response with headers (instead of just the body)

Reply:
Okay, no worries.